### PR TITLE
Fixes error message when Type is used as a Namespace

### LINKED ISSUE #12075: Incorrect error message for use of type as a namespace
```ts
interface OhNo {
}

declare let y: OhNo.hello;
//             ~~~~
//             'OhNo' refers to a type, but is being used as a value here.
```

`OhNo` is actually being used as a namespace, not a value. CC @jwbay in case you want to tackle this one. 😄

### PR DESCRIPTION
**Pull request by mylesmegyesi**
Fixes #12075

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Thanks for the fix!

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -1032,6 +1033,18 @@ namespace ts {
             }
         }
 
+        function checkAndReportErrorForUsingTypeAsNamespace(errorLocation: Node, name: string, meaning: SymbolFlags): boolean {
+            if (meaning === SymbolFlags.Namespace) {
+                const symbol = resolveSymbol(resolveName(errorLocation, name, SymbolFlags.Type & ~SymbolFlags.Value, /*nameNotFoundMessage*/undefined, /*nameArg*/ undefined));
```
Comment (amcasey) [inline src/compiler/checker.ts]: `SymbolFlags.Type & ~SymbolFlags.Value` excludes classes and enums.  Was that intentional? cc @DanielRosenwasser

Comment (mhegazy) [reply src/compiler/checker.ts]: I guess it should have been `& ~SymbolFlags.Namespace` instead

Comment (amcasey) [reply src/compiler/checker.ts]: I could be missing something, but I don't think `SymbolFlags.Type` includes `SymbolFlags.Namespace`.

**Pull request merged**